src/main/java/org/folio/circulation/resources/renewal/RenewalResource.java

```diff
@@ -189,6 +191,26 @@ private void renew(RoutingContext routingContext) {
       .thenApply(r -> r.map(this::toResponse))
       .thenAccept(webContext::writeResultToHttpResponse);
   }
+  
+  private Result<RenewalContext> unsetDateTruncationFlagIfNoOpenRecallsInQueue(
+    Result<RenewalContext> renewalContext) {
+    if (renewalContext.failed()) {
+      return renewalContext;
+    }
+
+    Loan loan = renewalContext.value().getLoan();
+    RequestQueue queue = renewalContext.value().getRequestQueue();
+
+    if (loan.wasDueDateChangedByRecall() && !hasOpenRecalls(queue)) {
+      loan.unsetDueDateChangedByRecall();
+    }
+    return renewalContext;
+  }
```

Comment:
I suggest changing this to use the result methods rather than checking for failed in a different flow, as it completely stops the flow of earlier failures

Reply:
Can you say more or demonstrate?  My java asynchronous programming skills are still pretty shaky.  I did try using the .map method, as you did in the other promise chain, but I could not get it to work-I kept getting compilation failures that indicated it wanted different arguments, and could not figure out how to shape the completionstage to a form it would accept.

Reply:
Shall I do something similar to last time where I provide a branch?

Reply:
Here is a branch with the change - https://github.com/folio-org/mod-circulation/tree/CIRC-1112-marc

If you'd like me to walk you through it, please let me know.

Reply:
Ah!  I see what I was doing wrong.  I was using the function correctly, I just wasn't binding the changed loan properly to the renewalcontext.  Obvious once pointed out.  Thank you.